packages/element/src/create-interpolate-element.js

```diff
@@ -0,0 +1,235 @@
+
+/**
+ * Transforms a given string value into a collection of html like tags.
+ *
+ * @example
+ *
+ * If you provide `something` as the tagString, this will return:
+ *
+ * ```js
+ * const tags = getTagsFromString( 'something' );
+ * const tagsAre = tags === [
+ *     '<something>',
+ *     '</something>',
+ *     '<something/>',
+ * ];
+ * ```
+ *
+ * @param {string} tagString  Something like `a%1`.
+ *
+ * @return {string[]} The generated tags for the given string.
+ */
+const getTagsFromString = ( tagString ) => {
+	return [
+		`<${ tagString }>`,
+		`</${ tagString }>`,
+		`<${ tagString }/>`,
+	];
+};
+
+/**
+ * Generates and returns the regular expression for the given type.
+ *
+ * @param {Array}		tags          Array of tags to use for the reg ex
+ * @param {string}  searchString  The search string serving as the base for the
+ *                                expression.
+ * @param {string}	type					What type of regEx to return.
+ *
+ * @return {RegExp}  The generated regular expression.
+ */
+const getRegEx = ( tags, searchString, type ) => {
+	let pattern;
+	const [ openTag, closeTag, selfClosingTag ] = tags;
+	switch ( type ) {
+		case 'children':
+			pattern = escapeRegExp( openTag ) + '(.*)' + escapeRegExp( closeTag );
+			break;
+		case 'propValue':
+			pattern = escapeRegExp( searchString );
+			break;
+		default:
+			pattern = escapeRegExp( selfClosingTag );
+	}
+	return new RegExp( pattern );
+};
+
+/**
+ * Generates and returns the regular expression used for splitting a string
+ *
+ * @param {string} searchString  The search string serving as the base for the
+ *                               expression.
+ *
+ * @return {RegExp}  The generated regular expression
+ */
+const getSplitRegEx = ( searchString ) => new RegExp( '(' + escapeRegExp( searchString ) + ')' );
+
+/**
+ * Generates a String.prototype.match value for the incoming arguments.
+ *
+ * @param {string} interpolatedString  The string the match is performed on.
+ * @param {string} searchString				 The string used as the base for the
+ *                                     search
+ * @param {Object} conversionConfig    The configuration object for the
+ *                                     interpolation being performed.
+ *
+ * @return {Array|null}	An array if there is a match or null if not.
+ */
+const getMatchFromString = (
+	interpolatedString,
+	searchString,
+	conversionConfig
+) => {
+	const tags = getTagsFromString( searchString );
+	// first try children reg ex.  If there is a match, then return.
+	const match = interpolatedString.match(
+		getRegEx( tags, searchString, 'children' )
+	);
+	if ( match !== null ) {
+		conversionConfig.hasChildren = true;
+		return match;
+	}
+	// no children, try matching on another pattern and return.
+	const regEx = getHasPropValue( conversionConfig ) ?
+		getRegEx( tags, searchString, 'propValue' ) :
+		getRegEx( tags, searchString );
+	return interpolatedString.match( regEx );
+};
```

Comment:
1. Maybe it's me, but the flow of first generating all possible `tags`, and then separately calling different versions of `getRegEx`, is hard to follow. Especially because `tags` is entirely derived from `searchString`, so strictly speaking passing both values to `getRegEx` is unnecessary and adds to the confusion. Do you think the following would be clearer?

```js
const expr = getBalancedTagsExpression( searchString );
const match = interpolatedString.match( expr );
…
const expr = escapeRegExp( searchString );
… // check for prop value match
const expr = getSelfClosingTagsExpression( searchString );
… // check for self-closing tag match
```

2. Perhaps a comment in the right place could shine light on why we check for prop value matches before self-closing ones. It makes sense, but is not immediately obvious.

Reply:
Done in e55ea05.